### ISSUE TITLE
Cannot enrol students to courses that have the turnitin plugin

### ISSUE THREAD

**Issue by christos312** [jmcgettrick/moodle-mod_turnitintooltwo#57]
Hello,

I was trying to enrol students on my Moodle installation and I was getting the following error.

 line 1368 of /lib/dml/mysqli_native_moodle_database.php: coding_exception thrown
* line 1424 of /lib/dml/mysqli_native_moodle_database.php: call to mysqli_native_moodle_database->update_record_raw()
* line 172 of /mod/turnitintooltwo/lib.php: call to mysqli_native_moodle_database->update_record()
* line 1219 of /lib/gradelib.php: call to turnitintooltwo_update_grades()
* line 1189 of /lib/gradelib.php: call to grade_update_mod_grades()
* line 1003 of /lib/gradelib.php: call to grade_grab_course_grades()
* line 1347 of /lib/enrollib.php: call to grade_recover_history_grades()
* line 160 of /enrol/manual/ajax.php: call to enrol_plugin->enrol_user()

The only way that I found to enrol students was to disabled turnitin plugin.

**Comment by mkpelletier** [comment]
Hello Christo,
Could you indicate which software versions you are using?  Which version of Moodle, which version of Turnitintooltwo?  We are using moodle with turnitintooltwo and have not had problems enrolling students.

**Comment by christos312** [comment]
Hello mkpelletier, 

I am using Moodle 2.9.2+, plugin version 2013111402

**Comment by mkpelletier** [comment]
Hello Christo,
Might I suggest that you update the plugin.  The version you are using is very old, and the issue you are experiencing is likely due to the fact that this version was not designed for use with Moodle 2.9.  There have been a host of fixes and improvements since that version was released.

Best of Luck!

**Comment by jmcgettrick** [comment]
Hi @christos312,
We have had  a couple of errors with this functionality recently which we've fixed. As @mkpelletier suggests, upgrading should take care of your error as it looks to be an error with updating the event in the Moodle calendar which we have fixed a while ago.

We have had a recent report of a problem which we've also fixed where grades were trying to be retrieved when enrolling a student. This is fixed by: https://github.com/jmcgettrick/moodle-mod_turnitintooltwo/pull/55/files.

Thanks for the input as well @mkpelletier :+1:
